### ISSUE TITLE
Add concurrent test cases

### ISSUE THREAD

**Issue by PragmaTwice** [apache/kvrocks-controller#215]
We need to add more cases about concurrency access from users, e.g. users can emit multiple migration tasks simultaneously from client.

**Comment by bseto** [comment]
I'm thinking of adding a `SlotOnly: false` test - in addition to the concurrency. 

For the `*redis.Client` connection in Node `ClusterNode`, would you prefer it connect to a real kvrocks, or is using a redis-mock ok?

redis mock from here: https://github.com/go-redis/redismock

**Comment by git-hulk** [comment]
> For the *redis.Client connection in Node ClusterNode, would you prefer it connect to a real kvrocks, or is using a redis-mock ok?


@bseto Yes, it would be nice to use Kvrocks, because some behaviors of Kvrocks is a bit different with Redis.

**Comment by Kaustubh1204** [comment]
I’m planning to work on this.

Approach:
- Add concurrent migration test cases where multiple clients emit migration tasks simultaneously.
- Focus on race conditions around sequence updates and slot state transitions.
- Use a real Kvrocks instance (via Docker) instead of redis-mock, to match production behavior differences noted above.
- Ensure tests can reliably reproduce ordering / concurrency edge cases.

I’ll start with minimal concurrent cases and expand coverage incrementally.
Feedback welcome before I open a PR.

**Comment by Kaustubh1204** [comment]
Hi @git-hulk , @PragmaTwice 

I’ve completed the implementation of the concurrent slot migration integration tests for issue #215. The tests cover multiple clients emitting migration tasks simultaneously, focusing on race conditions around sequence updates and slot state transitions.

The PR #376 is ready for your review. Feedback is welcome before merging.

Thanks!
